src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -112,10 +117,7 @@ public void onEvent(ResourceOffersEvent event, long sequence,
                     }
                 }
             } else {
-                LOGGER.info("No pending tasks, declining all offers");
-                for (Offer offer : offers) {
-                    driver.declineOffer(offer.getId());
-                }
+                offerLifecycleMgr.addOffers(offers);
             }
```

Comment:
Only calling `offerLifecyclerMgr.addOffers` when pendingTasks is empty seems like you could potentially loose offers you'd like to know about to add as dummy tasks.  Worst case: a task in pendingTasks that can't be fufilled and a back log of needed tasks to resize.

Side note: Processing offers in two locations in this manner is disorienting, is there a way to refactor, loop through offers once and assign as either type of task?